### ISSUE TITLE
Nested route with transition won't go away on navigation away

### ISSUE THREAD

**Issue by samal-rasmussen** [bluwy/svelte-router#3]
Hi and thanks for this router library. I am brand new to Svelte and I am looking for a router for my project that I am looking to build with Svelte. I really like the clean and simple api and functionality of this router, and would love to make it work for my use case. Unfortunately I am stumped on how to make a transitioning nested route go away again after having loaded it.

I have a working example of the functionality I am going for implemented with svelte-spa-router in the svelte repl:
https://svelte.dev/repl/7115e9ca03af42b88ed1f4d11e4ed64d?version=3.29.4

I wasn't able to get svelte-router to work in the repl, so I made the failing example in this commit:
https://github.com/samal-rasmussen/svelte-router/commit/2374229691a677a88f011e036f3bdbeaffa7b2b9

Steps to reproduce:
* Press Overlay
* And then press Main or the browser back button

As you can see the nested Overlay isn't going away when navigating away from it, and it seems that the Main component is being duplicated also if you look at the html.

**Comment by bluwy** [comment]
Thanks for reporting this! After some investigation, I found it was because Main.svelte was being re-rendered as Overlay.svelte was being transitioned out. This caused Svelte to freak out and re-rendered the Main twice.

`svelte-spa-router` handled this well because the navigations waited for a `tick()`, so the rendering was in order and not being interfered when transitioning. I thought about this awhile ago but didn't include a built-in tick (for more user control), perhaps it's a good reason now to implement it.

Another thing I want to tackle is the re-rendering of Main.svelte, the root route stayed the same so it shouldn't have triggered a re-render. `svelte-spa-router` suffers from this too.

I'll take some time to look into this in the weekends, might need to bump a minor release.

**Comment by bluwy** [comment]
UPDATE: Didn't get the finish the fix yesterday, Just when I thought I got it done, another problem comes up.

So far the solution was to use `<RouterView />` instead of slots for nested routes, breaking the simplicity of the router. Apparently svelte treats transitions in recursive components differently, causing the unmounting process to get "stucked" as in the repro.

But another issue rose, check this forked repl: https://svelte.dev/repl/38c7e56ef6e04fbd8aadcf3f88b89d94?version=3.29.4

Steps:
1. Click Overlay link
2. Click Diff link

When navigating from Overlay to Diff, the Main component immediately switches to Diff. What should theoretically happen is that Main component should not be unmounted until Overlay is unmounted (transition ends), according to how svelte works. But `svelte-spa-router` somehow "breaks" this rule.

Of course, breaking this rule seems to be the more desirable outcome, otherwise the Diff component would stack below Main while Main waits for Overlay to unmount.

I don't have a repro for this now, but I tried breaking this rule too, to no avail. I read `svelte-spa-router`'s code too, but nothing fishy seems to be going on there.

I have to delay this for awhile, it'll be a busy week for me. Perhaps I could rethink another way to tackle this.

Sorry you had to jump into the svelte router rabbithole, it was at least frustrating for me too when I started svelte and routers. Hopefully this will be solved when SvelteKit releases
